CMakeLists.txt

```diff
@@ -67,6 +68,8 @@ address, leak, undefined and thread.")
 option(PLUGIN_DENSE_PARSER "Build dense parser plugin" OFF)
 option(PLUGIN_RMM "Build with RAPIDS Memory Manager (RMM)" OFF)
 option(PLUGIN_FEDERATED "Build with Federated Learning" OFF)
+option(USE_GRPC_FROM_SYSTEM "Use gRPC from the system; if OFF, CMake will download
```

Comment:
Can we just leave the grpc installation to system package managers like conda or apt? I'm not entirely whether it's a good idea to maintain dependency with cmake ourselves, it's a build tool after all ...

Reply:
Conda decided to build `grpc-cpp` with GCC 11 (!), so it's incompatible with the compiler we use. 

The system package managers are a better bet. The only issue is that they'll probably give you an older version of gRPC.

@rongou Is it okay if we use an older version of gRPC?

Reply:
CentOS 7 yum: gRPC 1.20
Ubuntu 20.04 apt: gRPC 1.16

Reply:
Maybe we can just leave the grpc/protobuf installation to the CI script and keep the cmake file lean?

Reply:
Got it. I'll update the doc to show how to install gRPC in a Conda environment.

Reply:
We can probably make a list of (optional) build dependencies in our building from the source page and leave it for others to figure out how they want to install it. There are lots of package managers out there.

## Optional build dependencies (not runtime python deps, which is a much larger set):
- cudatoolkit
- nccl
- grpc
- OPENMP
- rmm
- nvtx
- HDFS
- S3, AZUER filesystem packages?
- ONEAPI ?
- MPI?
- sanitizers?

Reply:
Good idea.

As for which version of gRPC we should use in our CI, let's wait for @rongou's response

Reply:
I don't think we are using a lot of newer features in gRPC. The only issue is the CMake plugin in gRPC seems to be fairly recent (see https://www.f-ax.de/dev/2020/11/08/grpc-plugin-cmake-support.html). If you can get the code to build with older versions then it should be fine.

Reply:
I managed to get the code to build on conda env. Linking dynamic library failed for some reasons, but linking the static library works.